From 553725872c4738ceb63e653c990d3f8bfe6e480d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Sat, 26 Apr 2025 20:45:40 +0200
Subject: [PATCH] Numpy Patch

---
 executorlib/standalone/cache.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/executorlib/standalone/cache.py b/executorlib/standalone/cache.py
index dae1334e..8470e87a 100644
--- a/executorlib/standalone/cache.py
+++ b/executorlib/standalone/cache.py
@@ -1,7 +1,6 @@
 import os
 
 import cloudpickle
-import numpy as np
 
 group_dict = {
     "fn": "function",
@@ -25,6 +24,7 @@ def get_cache_data(cache_directory: str) -> list[dict]:
         list[dict]: List of dictionaries each representing on of the HDF5 files in the cache directory.
     """
     import h5py
+    import numpy as np
 
     file_lst = []
     for task_key in os.listdir(cache_directory):
